pkgs/development/libraries/highfive/default.nix

```
@@ -0,0 +1,43 @@
+{ stdenv
+, fetchFromGitHub
+, cmake
+, boost
+, eigen
+, hdf5
```

Another alternative which is commonly used
```suggestion
, hdf5
, mpiSupport ? hdf5.mpiSupport, mpi ? hdf5.mpi
```